TST: piecemeal updates to `test_base.py` for sparray conversion

dschult opened the pull request:
This first commit just changes .A to .toarray for places where we need sparray to not use .A due to deprecation of that feature.

perimosocordiae commented on scipy/sparse/tests/test_base.py:
```diff
@@ -2702,14 +2727,14 @@ def test_set_slice(self):
-    def test_assign_empty_spmatrix(self):
-        A = self.spmatrix(np.ones((2, 3)))
-        B = self.spmatrix((1, 2))
+    def test_assign_empty_spcreator(self):
```
This should probably be `test_assign_empty` or maybe `test_assign_empty_sparse`. Seems like possibly a find/replace error.

rgommers reviewed (approved):
Thanks @dschult, this looks good to me too. One comment inline, which isn't blocking.

In addition, I wanted to comment on the state of `test_base.py`. Tests are way too heavily parametrized, meaning that tests run slow and if a test fails it often fails with many duplicates. If you're going to work on these tests more, it'd be great if you could cut down on parametrization where you spot opportunities.

To illustrate, with `main` we now have:
```
$ python dev.py test -t scipy.sparse.tests.test_base
...
5617 passed, 1299 skipped, 9 deselected, 40 xfailed in 90.10s
```
And with this PR:
```
5663 passed, 1299 skipped, 9 deselected, 40 xfailed in 91.35s
```
Adding 46 tests and over a second in test suite execution time for what is just one simple 10 line long extra test case in this PR (`test_start_vs_at_sign_for_sparray_and_spmatrix`) is what that results in. That's not due to a problem with this PR, but it is excessive.

dschult commented:
I agree about the TODO comments. I've found a way to test *= for sparrays and fixed up that test.

And I see what you mean about a single test adding to the total. The single test in `_TestCommon` get repeated many times. I will keep an eye out for helping with this problem. I can imagine, e.g. that the test I added is really making sure that we understand how * and @ differ for the array and matrix versions of sparse. But I doubt that we need to check it for every dtype, for example. Of the total time testing, a lot is spent on Test64Bit. And it isn't even very much code, but the number of tests it generates is quite large. :)

rgommers reviewed (approved):
Update LGTM and CI is green modulo one unrelated failure in `optimize`. So in it goes, thanks @dschult
